File: fixtures/scale/dart/large/lib/ciphers/columnar/columnar_key.dart
class ColumnarKey {
  final String keyword;

  ColumnarKey(this.keyword);

  /// Column indexes sorted by their keyword letter, ties left to right.
  List<int> columnOrder() {
    final indexed = List.generate(keyword.length, (i) => i);
    indexed.sort((a, b) => keyword[a].compareTo(keyword[b]));
    return indexed;
  }

  /// Pads with X until the text fills whole rows.
  String padded(String text) {
    final sb = StringBuffer(text);
    while (sb.length % keyword.length != 0) sb.write('X');
    return sb.toString();
  }

  factory ColumnarKey.defaultKey() => ColumnarKey('ZEBRAS');
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/columnar/ColumnarKey.kt
package cryptobench.ciphers.columnar

data class ColumnarKey(val keyword: String) {

    /** Column indexes sorted by their keyword letter, ties left to right. */
    fun columnOrder(): IntArray =
        keyword.uppercase()
            .withIndex()
            .sortedWith(compareBy({ it.value }, { it.index }))
            .map { it.index }
            .toIntArray()

    /** Pads with X until the text fills whole rows. */
    fun padded(text: String): String {
        val sb = StringBuilder(text)
        while (sb.length % keyword.length != 0) sb.append('X')
        return sb.toString()
    }

    companion object {
        fun default(): ColumnarKey = ColumnarKey("ZEBRAS")
    }
}
